packages/salesforcedx-vscode-core/test/jest/decorators/scratch-org-decorator.test.ts

```diff
@@ -0,0 +1,80 @@
+import { ConfigUtil } from '@salesforce/salesforcedx-utils-vscode';
+import { mock } from 'sinon';
+import * as vscode from 'vscode';
+import URI from 'vscode-uri';
+import { monitorOrgConfigChanges, showOrg } from '../../../src/decorators';
+import { nls } from '../../../src/messages';
+
+describe('scratch org decorator', () => {
+  const testUser = 'test@username.com';
+  const browserIcon = `$(browser)`;
+  const openOrgCommand = 'sfdx.force.org.open';
+  const fakePathURI = URI.file('/Users/tester/mockNewFile.test.js');
+  const mockStatusBarItem = {
+    tooltip: '',
+    command: '',
+    show: jest.fn(),
+    text: ''
+  };
+  let onDidCreateEventEmitter: vscode.EventEmitter<vscode.Uri>;
+  let onDidChangeEventEmitter: vscode.EventEmitter<vscode.Uri>;
+  onDidCreateEventEmitter = new vscode.EventEmitter<vscode.Uri>();
+  onDidChangeEventEmitter = new vscode.EventEmitter<vscode.Uri>();
+  const mockWatcher = {
+    // onDidChange: onDidChangeEventEmitter.event,
+    // onDidCreate: onDidCreateEventEmitter.event
+    onDidChange: jest.fn(),
+    onDidCreate: jest.fn()
+  };
```

Comment:
You should do the mockWatcher assignment in the beforeEach so that the spys are reset between tests. Also mockStatusBarItem.show.  Basically you should always set spy values in the before* function.